x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/EmptyConfigUpdate.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public class EmptyConfigUpdate implements InferenceConfigUpdate {
+
+    public static final ParseField NAME = new ParseField("empty");
+
+    private static final ObjectParser<EmptyConfigUpdate, Void> PARSER =
+        new ObjectParser<>(NAME.getPreferredName(), EmptyConfigUpdate::new);
+
+    public static EmptyConfigUpdate fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public EmptyConfigUpdate() {
+    }
+
+    public EmptyConfigUpdate(StreamInput in) {
+    }
+
+    @Override
+    public InferenceConfig apply(InferenceConfig originalConfig) {
+        return originalConfig;
+    }
+
+    @Override
+    public InferenceConfig toConfig() {
+        return RegressionConfig.EMPTY_PARAMS;
+    }
```

Comment:
This will blow up if the user is attempting to infer against a classification model and the `InternalInferModelAction` has to serialize to a node before 7.8.0.

Reply:
Yikes we've got the same problem with the `ResultsFieldUpdate` added for the pipeline agg.

There is a work around but it won't be elegant

Reply:
@davidkyle The inference factory already knows the minimum node version. So, I suggest we use that to fail early when checking if this option can be left out or not.

Reply:
Good idea I've pushed that change